src/interfaces/products.ts

```diff
@@ -0,0 +1,30 @@
+import { z } from 'zod';
+
+export const createProductParams = z.object({
+  name: z.string(),
+  price: z.number(),
+  image: z.string(),
+  categoryId: z.number(),
+});
+export type CreateProductParams = z.infer<typeof createProductParams>;
+
+export const getProductParams = z.object({
+  id: z.string().transform((val) => parseInt(val)),
+});
+export type GetProductParams = z.infer<typeof getProductParams>;
+
+export const updateProductParams = z.object({
+  name: z.string(),
+  price: z.number(),
+  image: z.string(),
+});
+export type UpdateProductParams = z.infer<typeof updateProductParams>;
+
+export const getProductsParams = z.object({
+  name: z.string(),
+  categories: z.string().array(),
+  sortBy: z.string(),
+  page: z.string().transform((val) => parseInt(val)),
```

Comment:
I think is good you've divided your interface into separate files. But remember that to keep consistency in how we import things, you should create an index file and export everything from there